File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int total_ones = 0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='1'){
                total_ones++;
            }
        }
        int left_zero = 0; int right_ones = total_ones;
        int max_score = INT_MIN;
        for(int i=0;i<s.length()-1;i++){
            if(s[i]=='0'){
                left_zero+=1;
            }else{
                right_ones-=1;
            }
            
            int score = left_zero + right_ones;
            max_score = max(score, max_score);

        }
        return max_score;
    }
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution(object):
    def maxScore(self, s):
        """
        :type s: str
        :rtype: int
        """
        maxScore = left_zero = 0
        right_ones = s.count('1')

        for i in range(len(s)-1):
            left_zero += s[i] == '0'
            right_ones -= s[i] == '1'
            maxScore = max(maxScore, left_zero + right_ones) 
        return maxScore
        